smart-contracts/contracts/lp-strategy/src/ibc.rs

```diff
@@ -363,11 +369,13 @@ pub fn handle_icq_ack(
 ) -> Result<Response, ContractError> {
     // todo: query flows should be separated by which flowType we're doing (bond, unbond, startunbond)
 
-    let ack: InterchainQueryPacketAck = from_binary(&ack_bin)?;
-    let resp: CosmosResponse = CosmosResponse::decode(ack.data.0.as_ref())?;
+    let buf: InterchainQueryPacketAckData = from_json(ack_bin)?;
```

Comment:
Havn't gone through the internals, but look like in a new  version ack_bin is coming as json object? 
We can then call this variable as ack_json.

Reply:
Sure.